Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealitySpatialObserver.cs

```diff
@@ -0,0 +1,493 @@
+
+                    observer.Update(SurfaceObserver_OnSurfaceChanged);
+                    lastUpdated = Time.time;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Issue a request to the Surface Observer to begin baking the mesh.
+        /// </summary>
+        /// <param name="surfaceId">ID of the mesh to bake.</param>
+        private void RequestMesh(SurfaceId surfaceId)
+        {
+            string meshName = ("SpatialMesh - " + surfaceId.handle);
+
+            // todo:
+            SpatialMeshObject newMesh;
+            WorldAnchor worldAnchor;
+
+            if (!spareMeshObject.HasValue)
+            {
+                newMesh = CreateSpatialMeshObject(null, meshName, surfaceId.handle);
+
+                worldAnchor = newMesh.GameObject.AddComponent<WorldAnchor>();
+            }
+            else
+            {
+                newMesh = spareMeshObject.Value;
+                spareMeshObject = null;
+
+                newMesh.GameObject.name = meshName;
+                newMesh.Id = surfaceId.handle;
+
+                worldAnchor = newMesh.GameObject.GetComponent<WorldAnchor>();
+            }
+
+            Debug.Assert(worldAnchor != null);
+
+            SurfaceData surfaceData = new SurfaceData(
+                surfaceId,
+                newMesh.Filter,
+                worldAnchor,
+                newMesh.Collider,
+                MixedRealityManager.SpatialAwarenessSystem.MeshTrianglesPerCubicMeter,
+                true);
+
+            if (observer.RequestMeshAsync(surfaceData, SurfaceObserver_OnDataReady))
+            {
+                outstandingMeshObject = newMesh;
+            }
+            else
+            {
+                Debug.LogError($"Mesh request failed for Id == surfaceId.handle");
+                Debug.Assert(outstandingMeshObject == null);
+                ReclaimMeshObject(newMesh);
+            }
+        }
+
+        /// <summary>
+        /// Applys the configured observation extents.
+        /// </summary>
+        private void ConfigureObserverVolume()
+        {
+            Vector3 newExtents = MixedRealityManager.SpatialAwarenessSystem.ObservationExtents;
+
+            if (currentObserverExtents.Equals(newExtents)) { return; }
+            observer.SetVolumeAsAxisAlignedBox(Vector3.zero, newExtents);
+        }
+#endif // UNITY_WSA
+
+        /// <summary>
+        /// Handles the SurfaceObserver's OnSurfaceChanged event.
+        /// </summary>
+        /// <param name="id">The identifier assigned to the surface which has changed.</param>
+        /// <param name="changeType">The type of change that occurred on the surface.</param>
+        /// <param name="bounds">The bounds of the surface.</param>
+        /// <param name="updateTime">The date and time at which the change occurred.</param>
+        private void SurfaceObserver_OnSurfaceChanged(SurfaceId id, SurfaceChange changeType, Bounds bounds, System.DateTime updateTime)
```

Comment:
There are a few references to `SurfaceId` in non-`UNITY_WSA` wrapped code, which prevents building for non-UWP platforms.

![image](https://user-images.githubusercontent.com/3580640/47040829-1d9ef600-d13c-11e8-818c-a4e0f71d28e2.png)